### PR TITLE
[Bugfix] stripe query selection dropdown menu should display the selected component value on saving the query

### LINKED ISSUE #2104: [Stripe query] The selection dropdown menu should display the selected component value on saving the query
### What is the expected behaviour?
The selection dropdown menu should display the selected component value on saving the query

### What is the current behaviour?
The dropdown value is still empty even with a selected operation saved to the component state

### How to reproduce the issue?

1. Add stripe data source
2. Create a new stripe query
3. Select any operation from the selection dropdown
4. click save to save changes to the query

### Screenshots or Screencast
https://www.loom.com/share/13e5b35218164a5fbc1372fa8f249a60

* Update to `react-select` package from `react-select-search`
* Increase width of the selection component

### PR DESCRIPTION
**Pull request by arpitnath**
Fixes #2104

### CODE REVIEW

File: frontend/src/Editor/QueryManager/QueryEditors/Stripe.jsx [hunk 1]
```diff
@@ -96,20 +98,44 @@ class Stripe extends React.Component {
...
+    const newOptions = {
+      ...options,
+      params: {
+        ...options.params,
+        [paramType]: {
+          ...options.params[paramType],
+          [paramName]: undefined, // the value is undefined, it will be considered as removed as operation params are from the spec
+        },
+      },
+    };
```
Comment (gondar00) [inline frontend/src/Editor/QueryManager/QueryEditors/Stripe.jsx:112]:
```suggestion
    const newOptions = JSON.parse(JSON.stringify(options));
    options[paramName].= undefined;
```

**Review by gondar00 — CHANGES_REQUESTED** [review]

File: frontend/src/Editor/QueryManager/QueryEditors/Stripe.jsx [hunk 2]
```diff
@@ -96,20 +98,36 @@ class Stripe extends React.Component {
     this.props.optionsChanged(newOptions);
   };
 
-  renderOperationOption = (props, option, snapshot, className) => {
-    return (
-      <button {...props} className={className} type="button">
+  removeParam = (paramType, paramName) => {
+    const options = this.state.options;
+    const newOptions = JSON.parse(JSON.stringify(options));
+    options[paramName] = undefined;
```
Comment (gondar00) [inline frontend/src/Editor/QueryManager/QueryEditors/Stripe.jsx:104]: I think this should be `newOptions` ?
```suggestion
    newOptions[paramName] = undefined;
```

File: frontend/src/Editor/QueryManager/QueryEditors/Stripe.jsx [hunk 3]
```diff
@@ -96,20 +98,36 @@ class Stripe extends React.Component {
     this.props.optionsChanged(newOptions);
   };
 
-  renderOperationOption = (props, option, snapshot, className) => {
-    return (
-      <button {...props} className={className} type="button">
+  removeParam = (paramType, paramName) => {
+    const options = this.state.options;
+    const newOptions = JSON.parse(JSON.stringify(options));
```
Comment (gondar00) [inline frontend/src/Editor/QueryManager/QueryEditors/Stripe.jsx:103]: nit
```suggestion
    const newOptions = JSON.parse(JSON.stringify(this.state.options));
```

File: frontend/src/Editor/QueryManager/QueryEditors/Stripe.jsx [hunk 4]
```diff
@@ -96,20 +98,36 @@ class Stripe extends React.Component {
...
+    options[paramName] = undefined;
+
+    this.setState({
+      options: newOptions,
+    });
+
+    this.props.optionsChanged(newOptions);
```
Comment (gondar00) [inline frontend/src/Editor/QueryManager/QueryEditors/Stripe.jsx:110]: nit
```suggestion
    this.setState({
      options: newOptions,
    }, () => {
      this.props.optionsChanged(newOptions);
    });
```

**Review by gondar00 — CHANGES_REQUESTED** [review]

**Review by gondar00 — DISMISSED** [review]

**Comment by arpitnath** [comment]
Todo: Refactor

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**